flight/src/types.rs

```diff
@@ -0,0 +1,39 @@
+use cid::Cid;
+use serde::{Deserialize, Serialize};
+use serde_bytes::ByteBuf;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ConclusionEvent {
+    Data(ConclusionData),
+    Time(TimeEvent),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ConclusionInit {
+    pub stream_type: String,
+    pub controllers: String,
+    pub dimensions: Vec<(String, ByteBuf)>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ConclusionData {
+    pub id: Cid,
+    pub init: ConclusionInit,
+    // TODO : rethink this, add a checkpoint to make it work in datafusion query
+    pub previous: Vec<Cid>,
+    pub before: Option<CeramicTime>,
+    pub after: Option<CeramicTime>,
+    pub data: ByteBuf,
+}
+
+pub type CeramicTime = chrono::DateTime<chrono::Utc>;
+pub type Ipld = ByteBuf;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TimeEvent {
+    // WIP : Need clearer structure for time events
+    pub before: Option<CeramicTime>,
+    pub after: Option<CeramicTime>,
+    pub data: ByteBuf,
```

Comment:
What is the `data` for a time event? I don't think we need this one.

Reply:
Proof of anchor maybe?

Reply:
I think you just need the timestamp